operator/internal/controller/redpanda/redpanda_controller_integration_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build integration
```

Comment:
Same note about the build tag here but we can handle that after merging this PR.